lib/mirage.ml

```diff
@@ -1466,6 +1466,31 @@ module Substitutions = struct
     ] @ blocks @ networks
 end
 
+let opamify s =
+  let b = Buffer.create (String.length s) in
+  String.iter (function
+      | 'a'..'z' | 'A'..'Z' | '0'..'9'
+      | '_' | '-' | '+' as c -> Buffer.add_char b c
+      | '.' -> Buffer.add_char b '-'
+      | _ -> ()
+    ) s;
+  let s' = Buffer.contents b in
+  if String.length s' = 0 then raise (Invalid_argument s);
+  s'
+
+let long_name name target =
+  Fmt.strf "mirage-unikernel-%s-%a" (opamify name) Key.pp_target target
+
+let unikernel_name i target = match Info.output i with
+  | Some o -> opamify (Filename.basename o)
```

Comment:
When I configure with `-t ukvm` and `-o /otherdir/elsewhere`, should I get the `ukvm-bin` in the destination directory too?  I'll need it to do anything with the artifact, but it stays in the build directory currently:

```
~/mirage-skeleton/applications/static_website_tls $ mirage configure -t ukvm -o /home/user/unikernels/website
~/mirage-skeleton/applications/static_website_tls $ mirage configure -t ukvm -o /home/user/unikernels/website && make depend && make
~/mirage-skeleton/applications/static_website_tls$ ls ~/unikernel_apiary/
website.ukvm
```